metals/src/main/scala/scala/meta/internal/bsp/BspServers.scala

```diff
@@ -66,10 +67,16 @@ final class BspServers(
   ): Future[BuildServerConnection] = {
 
     def newConnection(): Future[SocketConnection] = {
-      scribe.info(s"Running BSP server ${details.getArgv}")
 
+      val args = details.getArgv.asScala.toList
+        /* Looks like we don't need to set it and it might sometimes break running BSP
+         * for example when we have sbt on path with whitespace.
+         */
+        .filterNot(arg => Properties.isWin && arg.contains("-Dsbt.script"))
```

Comment:
Huh, so is this never really needed?  Also these args are just created by sbt, if a user on windows has a space in their path and `sbt bspConfig` generates an `-Dsbt.script` with a space, and that's not valid, isn't that something that should be fixed on the sbt side? Or is it how we're using it that is actually the issue?

Reply:
This was the only way to make it work with older sbt :thinking: Maybe I should only do that for values with encoded special characters?

This seems to have been added in https://github.com/sbt/sbt/pull/6593 @adpi2 any idea why it might fail if sbt.script has an encoded space? Looks like Windows doesn't currently allow encoding, so maybe that's the issue?

Reply:
If you remove the `-Dsbt.script` option then the sbt server won't load the `.sbtopts` or the `.jvmopts` files which would be an issue in itself. Is there anything we can do on sbt side to encode/decode those white spaces properly?

Reply:
Last I checked is that you shouldn't actually encode those values, but it seems that when being used `-Dsbt.script` is not wrapped in "", which causes it not to work.

Reply:
I created https://github.com/sbt/sbt/issues/7199 and added it in my todo list.

Reply:
Thanks! I can test out the change on a Windows laptop if needed.

In the meantime, I think we can add a condition that would be more sure. I will push it in a bit.

Reply:
More info here https://learn.microsoft.com/en-us/troubleshoot/windows-client/networking/url-encoding-unc-paths-not-url-decoded

It seems that if you include `file:` that should work, but not sure everything will be able to read path as URI. Let me check

Reply:
Ok, that seems to work. I changed to that, we are able to work around without changing in sbt, though it might be useful to do same thing in sbt